students/ZidanLuo/session03/mailroom1.py

```diff
@@ -0,0 +1,99 @@
+import sys
+
+#Create a list of data of donors
+donors = [("Fred Jones", [100, 20]), 
+("Amy Shumer", [2000, 99]),
+("Yu Wang",[5]),
+("Jin Liu",[520]),
+("Doron Levy",[3]) ]
+
+#prompt structure learned from the mailroom tutorial
+prompt = "\n".join(("Welcome to the mailroom!", 
+	      "please choose from the following:", 
+	      "t - Thank You Email", 
+	      "c - Create a Report",
+	      "q - Quit", 
+	      ">>>  "))
+
+def get_length_a():
+	ls = []
+	ls2 = []
+	for donor in donors:
+		i = len(str(sum(donor[1])))
+		ls.append(i)
+		ls2.append(len(donor[0]))
+		width1 = max(ls2) + 8
+		width2 = max(ls) + 8
+	return width1, width2
+
+def in_list(donor_name):
+	for names in donors:
+		if donor_name in names:
+			return True
+	return False
+
+def update_donor(response_donor, response_amount):
+	index = 0
+	for names in donors:
+		if response_donor in names:
+			index = donors.index(names)
+			break
+	lst = list(donors[index])
+	lst[1].append(int(response_amount))
+	#massages sent in the email
+	print('{:20}'.format("Dear " + response_donor + ":"))
+	print("Thank you for donating " + response_amount + "!")
+	print()
+
+def thank_you():
+	while True:
+		response_donor = input("Please provide the donar's full name or type list to see the full history: \n")
+		if response_donor == "list":
+			print(donors)
+		else:
+			response_amount = input("Please provide the donation amount: \n")
+			if in_list(response_donor):
+				update_donor(response_donor, response_amount)
+				break
+			else:
+				donors.append((response_donor,[int(response_amount)]))
+				print('{:20}'.format("Dear " + response_donor + ":"))
+				print("Thank you for donating " + response_amount + "!")
+				print()
+				break
```

Comment:
This works well, but will create a new donor if first or last name is not capitalized.  Might want to account for that.